src/power_grid_model/core/data_handling.py

```diff
@@ -184,7 +184,8 @@ def process_output_component_types(
     for comp_name, attrs in output_component_types.items():
         if attrs is None:
             continue
-        diff = set(attrs).difference(output_meta[comp_name].dtype.names)
+        attr_names = output_meta[comp_name].dtype.names
+        diff = set(attrs).difference(attr_names) if attr_names is not None else set(attrs)
```

Comment:
it **should** never be `None` in our case but NumPy disagrees. Using `assert`s in Python in production code is bad practice, because it suggests test code in production code to most people. IMO it can't hurt to have an additional `None` check